enterprise/server/invocation_stat_service/invocation_stat_service.go

```diff
@@ -135,79 +140,93 @@ func flattenTrendsQuery(innerQuery string) string {
 	FROM (` + innerQuery + ")"
 }
 
-func addWhereClauses(q *query_builder.Query, req *stpb.GetTrendRequest) error {
-	groupID := req.GetRequestContext().GetGroupId()
+func addWhereClauses(q *query_builder.Query, tq *stpb.TrendQuery, reqCtx *ctxpb.RequestContext, lookback int32) error {
 
-	if user := req.GetQuery().GetUser(); user != "" {
+	if user := tq.GetUser(); user != "" {
 		q.AddWhereClause("user = ?", user)
 	}
 
-	if host := req.GetQuery().GetHost(); host != "" {
+	if host := tq.GetHost(); host != "" {
 		q.AddWhereClause("host = ?", host)
 	}
 
-	if repoURL := req.GetQuery().GetRepoUrl(); repoURL != "" {
+	if repoURL := tq.GetRepoUrl(); repoURL != "" {
 		q.AddWhereClause("repo_url = ?", repoURL)
 	}
 
-	if branchName := req.GetQuery().GetBranchName(); branchName != "" {
+	if branchName := tq.GetBranchName(); branchName != "" {
 		q.AddWhereClause("branch_name = ?", branchName)
 	}
 
-	if command := req.GetQuery().GetCommand(); command != "" {
+	if command := tq.GetCommand(); command != "" {
 		q.AddWhereClause("command = ?", command)
 	}
 
-	if commitSHA := req.GetQuery().GetCommitSha(); commitSHA != "" {
+	if commitSHA := tq.GetCommitSha(); commitSHA != "" {
 		q.AddWhereClause("commit_sha = ?", commitSHA)
 	}
 
 	roleClauses := query_builder.OrClauses{}
-	for _, role := range req.GetQuery().GetRole() {
+	for _, role := range tq.GetRole() {
 		roleClauses.AddOr("role = ?", role)
 	}
 	if roleQuery, roleArgs := roleClauses.Build(); roleQuery != "" {
 		q.AddWhereClause("("+roleQuery+")", roleArgs...)
 	}
 
-	if start := req.GetQuery().GetUpdatedAfter(); start.IsValid() {
+	if start := tq.GetUpdatedAfter(); start.IsValid() {
 		q.AddWhereClause("updated_at_usec >= ?", start.AsTime().UnixMicro())
 	} else {
 		// If no start time specified, respect the lookback window field if set,
 		// or default to 7 days.
 		// TODO(bduffany): Delete this once clients no longer need it.
 		lookbackWindowDays := 7 * 24 * time.Hour
-		if w := req.GetLookbackWindowDays(); w != 0 {
-			if w < 1 || w > 365 {
+		if lookback != 0 {
```

Comment:
nit: is the unit of lookback day? and lookbackWindowDays is actually hours? can we rename the two names to clarify the units?

Reply:
thanks, i fixed up the units on these (the "days" field is a duration anyway)